### PR TITLE
feat(docs): update inline documentation in experimental, testing, timezone, and types packages

### PR DESCRIPTION
**Pull request by sanderson**
Updates the inline documentation in the following packages:

- experimental
- testing
- timezone
- types

### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: stdlib/testing/testing.flux [hunk 1]
```diff
@@ -1,108 +1,131 @@
...
+// The function matches tables from each stream based on group keys.
+// For each matched table, it produces a diff.
+// Any added or removed rows are added to the table as a row.
+// An additional string column with the name diff is created and contains a
+// `-` if the row was present in the `got` table and not in the `want` table or
+// `+` if the opposite is true.
 //
-// The diff function is guaranteed to emit at least one row if the tables are different and no rows if the tables are the same. The exact diff produced may change.
-// diff can be used to perform in-line diffs in a query.
+// `diff()` function is guaranteed to emit at least one row if the tables are
```
Comment (noramullen1) [inline stdlib/testing/testing.flux:91]:
```suggestion
// `diff()` function emits at least one row if the tables are
```

File: stdlib/testing/testing.flux [hunk 2]
```diff
@@ -268,13 +320,16 @@ inspect = (case) => {
 // run executes a specified test case.
 //
 // ## Parameters
-// - `case` is the test case to run.
+// - case: Test case to run.
+//
+// ## Exmamples
```
Comment (noramullen1) [inline stdlib/testing/testing.flux:325]:
```suggestion
// ## Examples
```

**Review by noramullen1 — APPROVED** [review]
Just a couple small changes

**Comment by skartikey** [comment]
@sanderson The PR for [adding location option support to date package](https://github.com/influxdata/flux/pull/4327) is merged yesterday. As the changes in this PR are in the same area you might want to revisit your changes. Kindly let me know if you need any help.

**Comment by sanderson** [comment]
@skartikey Is the default for the `location` param in the `date` package the value of the `universe.location` option?

**Comment by skartikey** [comment]
> @skartikey Is the default for the `location` param in the `date` package the value of the `universe.location` option?

with the new change, the location option has been moved from the universe package to the prelude(Prelude are the names and types of values that are in-scope in all Flux scripts.). The default behavior is to return a linear clock and no offset. You can find the location option declaration [here](https://github.com/influxdata/flux/blob/master/stdlib/internal/location/location.flux).

**Pull request merged**